src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js

```diff
@@ -791,6 +791,68 @@ describe('ReactDOMComponent', () => {
       );
     });
 
+    it('should emit a warning once for a named custom component using shady DOM', () => {
+      if (ReactDOMFeatureFlags.useCreateElement) {
+        spyOn(console, 'error');
+
+        var defaultCreateElement = document.createElement.bind(document);
+
+        try {
+          document.createElement = element => {
+            var container = defaultCreateElement(element);
+            container.shadyRoot = {};
+            return container;
+          };
+          var ShadyComponent = React.createClass({
+            render() {
+              return <div { ...this.props }></div>;
```

Comment:
Rendering just `<div />` fails the `isCustomComponentTag` check, which was the reason why props were being passed to simulate an `is=` custom component.

I found a remedy to this by rendering anything with a hyphen in the name (which happens to be the naming convention of all Polymer components) like `<polymer-component />` instead. No props needed!